docs: Add details on production deployments (#297)
navapbc/labs-decision-support-tool: docs/releases.md

# Deployments

The pilot instance of the chatbot has two environments, `DEV` and `PROD`.

Dev is a CD environment; merges to `main` automatically trigger deploys.

## Deploying to production

Prod requires a manual deploy step:
 1. On the [Releases page](https://github.com/navapbc/labs-decision-support-tool/releases), select `Draft a new release`

>  1. Under `Choose a tag`, create a new tag, bumping the version number as appropriate (e.g., `v1.4.0` if the previous version was `v1.3.0`)

Reviewer: Should we provide some guidance on version numbers, like semantic versioning?

Author: 👍 Good suggestion, will add

 1. Select `Generate release notes` to pre-populate the rest of the form. Adjust the notes as needed
 1. Click `Publish release` 
 1. [Select the Deploy App GitHub Action](https://github.com/navapbc/labs-decision-support-tool/actions/workflows/cd-app.yml)
 1. Under `Environment to deploy to`, select `prod`
 1. Under `Tag or branch or SHA to deploy`, enter the tag you created (e.g., `v1.4.0`)
 1. Click `Run workflow` (do not change `Use workflow from`)

> 

Author: ```suggestion

We follow [semantic versioning](https://semver.org/) for version numbers.

```

# Release Management

> There are a few makefile commands that can be used to automate cutting and deploying releases. We don't use these on our pilot.

Reviewer: Is this suggestion true?
```suggestion
There are a few makefile commands that can be used to automate cutting and deploying releases. We don't use the following on our pilot.
```

If so, should the instructions be removed or hidden?

Author: Ah, what I meant was we don't need *manually* use them (the github actions take care of this under the hood) -- the original explanations here felt to me like the implied we needed to run these commands manually. Let me revise to clarify that

# Deployments

The pilot instance of the chatbot has two environments, `DEV` and `PROD`.

Dev is a CD environment; merges to `main` automatically trigger deploys.

## Deploying to production

Prod requires a manual deploy step:
 1. On the [Releases page](https://github.com/navapbc/labs-decision-support-tool/releases), select `Draft a new release`

>  1. Under `Choose a tag`, create a new tag, bumping the version number as appropriate (e.g., `v1.4.0` if the previous version was `v1.3.0`)

Author: ```suggestion
 1. Under `Choose a tag`, create a new tag, bumping the version number as appropriate to follow [semantic versioning](https://semver.org/)
```
